Framework/PythonInterface/plugins/algorithms/WorkflowAlgorithms/SANSILLMultiProcess.py

```diff
@@ -0,0 +1,990 @@
+            return [empty_beam_ws, empty_beam_flux]
+        else:
+            return ['','']
+
+    def process_flux(self, d, dark_current_ws):
+        '''Processes the empty beam for flux at the given distance'''
+        runs = self.getPropertyValue('FluxRuns')
+        if runs:
+            empty_beam = runs.split(',')[d]
+            [process_empty_beam, empty_beam_ws] = needs_processing(empty_beam, 'EmptyBeam')
+            empty_beam_flux = empty_beam_ws + 'Flux'
+            if process_empty_beam:
+                SANSILLReduction(Runs=empty_beam,
+                                 ProcessAs='EmptyBeam',
+                                 DarkCurrentWorkspace=dark_current_ws,
+                                 NormaliseBy=self.getProperty('NormaliseBy').value,
+                                 TransmissionBeamRadius=self.get_tr_beam_radius(self.tr_index(d)),
+                                 OutputWorkspace=empty_beam_ws,
+                                 OutputFluxWorkspace=empty_beam_flux)
+            return [empty_beam_ws, empty_beam_flux]
+        else:
+            return ['','']
+
+    def process_container(self, d, dark_current_ws, empty_beam_ws, transmissions):
+        '''Processes the empty container at the given distance'''
+        runs = self.getPropertyValue('EmptyContainerRuns')
+        if runs:
+            empty_can = runs.split(',')[d]
+            [process_empty_can, empty_can_ws] = needs_processing(empty_can, 'EmptyContainer')
+            if process_empty_can:
+                can_tr_ws = ''
+                if transmissions and 'Container' in transmissions[self.tr_index(d)]:
+                    can_tr_ws = transmissions[self.tr_index(d)]['Container']
+                SANSILLReduction(Runs=empty_can,
+                                 ProcessAs='EmptyContainer',
+                                 DarkCurrentWorkspace=dark_current_ws,
+                                 EmptyBeamWorkspace=empty_beam_ws,
+                                 TransmissionWorkspace=can_tr_ws,
+                                 TransmissionThetaDependent=self.getProperty('TransmissionThetaDependent').value,
+                                 NormaliseBy=self.getProperty('NormaliseBy').value,
+                                 OutputWorkspace=empty_can_ws)
+            return empty_can_ws
+        else:
+            return ''
+
+    def process_sample(self, d, dark_current_ws, empty_beam_ws, empty_can_ws, flux_ws, transmissions):
+        '''Processes all the samples at the given distance'''
+        runs = self.getPropertyValue(f'SampleRunsD{d+1}')
+        if runs:
+            [edge_mask_ws, beam_stop_mask_ws] = self.load_masks(d)
+            flat_field_ws = self.load_flat_field(d)
+            solvent_ws = self.load_solvent(d)
+            sens_ws = self.load_sensitivity(d)
+            sample_tr_ws = transmissions[self.tr_index(d)]['SampleTransmission'] if transmissions else ''
```

Comment:
If I run the algorithm with SampleRunsD1 populated and SampleTrRunsW2 populated (but SampleTrRunsW1 empty) I get an error from this line because transmissions =[{}] and this evaluates to true because it's a non-empty list

Reply:
Transmission runs as well have to be populated from W1 onwards. If there is only one wavelength, they have to be filled at W1, I'll add a validation for this, too.